### PR TITLE
ESDT improvements fixes

### PR DESCRIPTION
**Pull request by miiu96**
- Fixes
- Added log events for the new builtin functions

### CODE REVIEW

File: builtInFunctions/esdtMetaDataUpdate.go [hunk 1]
```diff
@@ -142,6 +149,14 @@ func (e *esdtMetaDataUpdate) ProcessBuiltinFunction(acntSnd, _ vmcommon.UserAcco
 		ReturnCode:   vmcommon.Ok,
 		GasRemaining: vmInput.GasProvided - gasToUse,
 	}
+
+	esdtDataBytes, err := e.marshaller.Marshal(esdtInfo.esdtData)
+	if err != nil {
+		log.Warn("esdtMetaDataUpdate.ProcessBuiltinFunction: cannot marshall esdt data for log", "error", err)
+	}
+
+	addESDTEntryInVMOutput(vmOutput, []byte(core.ESDTMetaDataUpdate), vmInput.Arguments[0], esdtInfo.esdtData.TokenMetaData.Nonce, big.NewInt(0), vmInput.CallerAddr, esdtDataBytes)
```
Comment (ssd04) [inline builtInFunctions/esdtMetaDataUpdate.go:158]: is it safe to pass `esdtDataBytes` as nil here in case of marshall error?

Comment (miiu96) [reply builtInFunctions/esdtMetaDataUpdate.go:158]: This should never happen

**Review by sstanculeanu — APPROVED** [review]

File: builtInFunctions/esdtMetaDataUpdate.go (hunk 1 above)
Comment (BeniaminDrasovean) [reply builtInFunctions/esdtMetaDataUpdate.go:158]: Checked `addESDTEntryInVMOutput` and there should be no problem even if `esdtDataBytes` is nil

File: parsers/dataField/utils.go [hunk 2]
```diff
@@ -56,6 +56,11 @@ func getAllBuiltInFunctions() []string {
 		core.BuiltInFunctionSetGuardian,
 		core.BuiltInFunctionUnGuardAccount,
 		core.BuiltInFunctionGuardAccount,
+		core.ESDTMetaDataRecreate,
+		core.ESDTMetaDataUpdate,
+		core.ESDTSetNewURIs,
+		core.ESDTModifyCreator,
+		core.ESDTModifyRoyalties,
```
Comment (BeniaminDrasovean) [inline parsers/dataField/utils.go:63]: Should `core.ESDTSetTokenType` also be added?

Comment (miiu96) [reply parsers/dataField/utils.go:63]: added

**Review by BeniaminDrasovean — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

**Pull request merged**